pkg/operation/seed/seed.go

```diff
@@ -864,3 +894,90 @@ func deleteMutatingWebHookConfiguration(ctx context.Context, k8sSeedClient clien
 	err := k8sSeedClient.Delete(ctx, a)
 	return client.IgnoreNotFound(err)
 }
+
+func getIngressClass(seedNginxEnabled bool) string {
+	if seedNginxEnabled {
+		return v1beta1constants.SeedNginxIngressClass
+	}
+	return "nginx"
+}
+
+func migrateIngressClassForShootIngresses(ctx context.Context, seedClient client.Client) error {
+	shootNamespaces := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, shootNamespaces, client.MatchingLabels{"gardener.cloud/role": "shoot"}); err != nil {
+		return err
+	}
+	for _, ns := range shootNamespaces.Items {
+		if err := switchIngressClassToSeedNginx(ctx, kutil.Key(ns.Name, "prometheus"), seedClient); err != nil {
+			return err
+		}
+		if err := switchIngressClassToSeedNginx(ctx, kutil.Key(ns.Name, "grafana-operators"), seedClient); err != nil {
+			return err
+		}
+		if err := switchIngressClassToSeedNginx(ctx, kutil.Key(ns.Name, "grafana-users"), seedClient); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func switchIngressClassToSeedNginx(ctx context.Context, ingressKey types.NamespacedName, seedClient client.Client) error {
+	ingress := &extensionsv1beta1.Ingress{}
+	if err := seedClient.Get(ctx, ingressKey, ingress); err != nil {
```

Comment:
If we receive a "not found" error here then we shouldn't do anything. Some shoots might not have these ingresses, and that's perfectly fine for us.